### DISCUSSION TITLE
How to safely update to a new version?

### DISCUSSION THREAD

**Discussion by stamminator** [LesFerch/WinSetView#120] [Q&A] [answered]
I'm running version 2.96 and want to update to 2.99. If I delete the older version's directory and swap in the new one, I'm assuming I'll lose my configuration settings (e.g. AppData/MySettings.ini). Is it safe to just swap in the new WinSetView.exe into the existing directory? Or should I copy certain files from the old version's directory to the new one, then do a total swap?

As a side note, I love the portable aspect of this app, but having an installer than does this process automatically would be very nice.

Thanks for making this great app! I recommend it to all my friends who are Windows power users.

**Comment by LesFerch** [comment, marked as answer]
You can copy the entire 2.99 zip contents to your 2.96 folder, replacing what's there. The AppData folder (where your settings are stored) will not be overwritten. The 2.99 zip does not include an AppData folder. It's created on the fly if it does not already exist. These changes, to protect your settings, was a recent enhancement (post 2.96).

Note that if you put WinSetView in a protected location, such as Program Files, the active settings will be in your %LocalAppData% folder instead of the WinSetView\AppData folder.

Some day there will likely be a WinGet (and possibly a Chocolatey) package. A user did create a WinGet package, but there was an issue that I tracked down to their use of a symlink. If I can get them to use a shortcut instead of the symlink it should work fine. For me, this is a flash drive tool, but I try to make everyone happy when I can. 🙂

P.S. A tool that I live by is [Beyond Compare](https://www.scootersoftware.com/) (but the free [WinMerge ](https://winmerge.org/?lang=en)is great too). Using a tool like that, you can compare the old and new folders and immediately see the differences.

**Reply by stamminator** [reply]
Thank you! Worked like a charm. Might be worth putting a blurb about updating in the README.

**Reply by LesFerch** [reply]
Will do. Thanks for the suggestion. 👍
